cmd/eval.go

```diff
@@ -249,3 +266,60 @@ func vsGoldenHtmlFn(cmd *cobra.Command, args []string) {
 		log.Error(err)
 	}
 }
+
+func testTypoSuggestFn(cmd *cobra.Command, args []string) {
+	esManager := search.MakeESManager(elasticUrl)
+	esc, err := esManager.GetClient()
+	utils.Must(err)
+	engine := search.NewESEngine(esc, nil, nil, nil, nil)
+
+	evalSet, err := search.InitAndReadEvalSet(evalSetPath)
+	utils.Must(err)
+
+	file, err := os.Open(typosPath)
+	utils.Must(err)
+	defer file.Close()
+
+	typos := make([]string, 0)
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		typos = append(typos, scanner.Text())
+	}
+	utils.Must(scanner.Err())
+
+	falsePositiveCnt := 0
+	noSuggestForTypoCnt := 0
+
+	log.Info("*** CHECKING KNOWN TYPOS ***")
+
+	for _, t := range typos {
+		query := search.Query{Term: t, LanguageOrder: consts.SEARCH_LANG_ORDER[language]}
+		res, err := engine.GetTypoSuggest(query)
+		utils.Must(err)
+		if res.Valid {
+			log.Infof("Suggest for '%s' is: '%s'.", t, res.String)
+		} else {
+			noSuggestForTypoCnt++
+			log.Infof("No suggest for '%s'!", t)
+		}
+	}
+
+	log.Info("\n*** CHECKING FALSE POSITIVE FROM RECALL SET***")
+
+	for _, e := range evalSet {
+
+		query := search.Query{Term: e.Query, LanguageOrder: consts.SEARCH_LANG_ORDER[language]}
+
+		res, err := engine.GetTypoSuggest(query)
+		utils.Must(err)
+		if res.Valid {
+			log.Infof("Suggest for '%s' is: '%s'. Check if this is false positive.", e.Query, res.String)
```

Comment:
Why falsePositiveCnt?
Is this because eval set should have no typos?
I think that is not 100% true... true in most cases though...